### ISSUE TITLE
`Tensor.scatter_` unclear documentation

### ISSUE THREAD

**Issue by mattbarrett98** [pytorch/pytorch#119543]
### 📚 The doc issue

The following code works:

```python
import torch

x = torch.tensor([1, 2])
index=torch.tensor([0, 1])

x.scatter_(dim=0, index=index, value=0)
print(x)  # tensor([0, 0])
```
`torch.Tensor.scatter_` allows the use of a `value` keyword arg which is undocumented https://pytorch.org/docs/stable/generated/torch.Tensor.scatter_.html. 

It seems this is the mechanism through which `src` is able to allow floats, as well as tensors. But this would imply someone could do something like 
```python
x.scatter_(dim=dim, index=index, src=0.0)
```
Though this will fail, with an error saying `src` needs to be a tensor, contradicting the docs for this method.

### Suggest a potential alternative/fix

The docs could be a bit clearer, mainly in reference to the parameter type hint `src (Tensor or float)`, which isn't entirely true.

cc @svekars @brycebortree @mikaylagawarecki

**Comment by vadimkantorov** [comment]
Related: https://github.com/pytorch/pytorch/issues/5405

**Comment by janeyx99** [comment]
cc @mikaylagawarecki for scatter gather

**Comment by mikaylagawarecki** [comment]
@atskae Sure, we could check whether we could bugfix this by changing `value -> src` in these three signatures.

If that does not work, I think the best way to clarify this would be to add a second signature like we do [in the docs for `torch.sum`](https://pytorch.org/docs/stable/generated/torch.sum.html)

So first signature would be
`Tensor.scatter_(dim, index, src, reduce=None) → Tensor`
and second signature would be 
`Tensor.scatter_(dim, index, value, reduce=None) → Tensor`

with the appropriate argument annotation

**Comment by atskae** [comment]
> @atskae Sure, we could check whether we could bugfix this by changing value -> src in these three signatures.

@mikaylagawarecki where could I find the three signatures in the source code? I was only able to find one definition of `scatter_()` in [`lowering.py`](https://github.com/pytorch/pytorch/blob/87c6cd2f0069bb8ef990098266b916f3e6984e3f/torch/_inductor/lowering.py#L3235). I'm not sure if the `*` in the signature would include the `value` keyword, but `*` is also not passed down to `scatter_fallback()` or `scatter_reduce_()`.
